### ISSUE TITLE
HotColumn data prop definition does not include string

### ISSUE THREAD

**Issue by yzaroui** [handsontable/handsontable#7612]
### Description
When using an object as data source as mentioned in the documentation here :
https://handsontable.com/docs/8.3.1/frameworks-wrapper-for-react-hot-column.html#object-data-source
Typescript is not happy because data is of type `data?: CellValue[][] | RowObject[];` and can't take a string value like in the example : 

```javascript
<HotTable settings={this.state.hotSettings}>
  {/* use data prop to make a reference to the column data*/}
  <HotColumn data="id" />
  <HotColumn data="name" />
  <HotColumn data="score">
    <ScoreRenderer hot-renderer />
  </HotColumn>
  <HotColumn data="isPromoted">
    <PromotionRenderer hot-renderer />
  </HotColumn>
</HotTable>
```

### Your environment
* Handsontable version: 8.3.1
* Browser Name and version: Brave latest
* Operating System: Win 10

**Comment by AMBudnik** [comment]
Thank you for sharing @yzaroui 

Related topics:
* https://github.com/handsontable/docs/issues/189 
* https://github.com/handsontable/docs/issues/174

**Comment by mealplana** [comment]
I have the same problem in 8.3.2. The `data` property accepts a string and the code works with a string, but it seems the Typescript definitions are not correct. I am ignoring one line of TS for now as a workaround with:
```
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
```

but ideally the type definition would be updated. As far as I understand, and unless I'm missing something, this should be a really easy fix. Just replacing `data?: CellValue[][] | RowObject[];` with `data?: CellValue[][] | RowObject[] | string;`

**Comment by AMBudnik** [comment]
Hi @yzaroui @mealplana 

we just released Handsontable v8.4.0 where this issue is solved.

Please update your version and let me know if it works for you as well.
